packages/enketo-express/app/lib/db.js

```diff
@@ -1,21 +1,26 @@
 const redis = require('redis');
 const config = require('../models/config-model').server;
 
-const mainClient = redis.createClient(
-    config.redis.main.port,
-    config.redis.main.host,
-    {
-        auth_pass: config.redis.main.password,
-    }
-);
+/**
+ * Creates the redis url passed into the client, taking the URL as is if defined, otherwise building the url from
+ * the predefined host, port and option auth token
+ *
+ * @static
+ * @param { object } redisConfig - Local redis configuration settings imported from the config model.
+ * @return { string } Redis url
+ */
 
-const cacheClient = redis.createClient(
-    config.redis.cache.port,
-    config.redis.cache.host,
-    {
-        auth_pass: config.redis.cache.password,
+function _getRedisUrl(redisConfig) {
+    if (redisConfig.url) {
+        return redisConfig.url
+    }
+    else {
+        return `redis://:${redisConfig.password}@${redisConfig.host}:${redisConfig.port}`
```

Comment:
If password is `null` in the env/config, please exclude the `:${redisConfig.password}@` part of the redis URL. As-is, there's a warning produced.